### ISSUE TITLE
F-Droid needs Antifeature Tracking

### ISSUE THREAD

**Issue by licaon-kter** [ooni/probe#1391]
...as usage metrics and send error reports are OPT-OUT

Also, the app connects to `countly.ooni.io` port 443 TCPv4 and `oXXXXXX.ingest.sentry.io` port 443 TCPv4 just by starting the app, no test ever run. 

Making them OPT-IN is better ;)

F-Droid version 2.9.10

**Comment by lorenzoPrimi** [comment]
Countly is already opt in in the last version,
if you check the `register` function
https://github.com/ooni/probe-android/blob/master/app/src/main/java/org/openobservatory/ooniprobe/common/CountlyManager.java#L35
it has 
`.setRequiresConsent(true)`
and the consents are initialized to false.

Same thing happen with sentry, it just init the sdk but before sending any crash it checks the permission
https://github.com/ooni/probe-android/blob/master/app/src/main/java/org/openobservatory/ooniprobe/common/Application.java#L56

So even if you see any outgoing connection is not sending any data to these platform.

**Comment by licaon-kter** [comment]
Both toggles are ON **in the app**.

The app connects to those servers, thus leaking IP/user-agent/etc.

**Comment by lorenzoPrimi** [comment]
The toggles are changed to ON only when you click "default settings", they are OFF by default. 
https://github.com/ooni/probe-android/blob/master/app/src/main/java/org/openobservatory/ooniprobe/common/PreferenceManager.java#L90

Tap change settings instead of default settings and you will see that defaults is off for both.

xRef:
https://github.com/ooni/probe-android/blob/master/app/src/main/java/org/openobservatory/ooniprobe/common/PreferenceManager.java#L69
https://github.com/ooni/probe-android/blob/master/app/src/main/res/xml/preferences_global.xml#L288
fdroid: https://github.com/ooni/probe-android/blob/master/app/src/fdroid/res/xml/preferences_global.xml#L278

**Comment by licaon-kter** [comment]
> The toggles are changed to ON only when you click "default settings", they are OFF by default.

Where is this exactly?

I start the app, press "I understand" twice, answer 2 quiz questions (Correctly), press "Let's start".

Then go to Settings, Privacy... toggles are ON.

**Comment by licaon-kter** [comment]
Oh....there's something.... let me open another issue first

**Comment by lorenzoPrimi** [comment]
There are two buttons, one it's "Let's go" and another is titled "Change defaults"
if you press "Let's go" you accept the default settings that are listed above, in that moment the default will be set, not before.
if you press "Change defaults" and go to the settings page you will see that the send crash and send analytics are OFF

**Comment by licaon-kter** [comment]
Ok, so there are... in the second picture: https://github.com/ooni/probe/issues/1392

But still applies
> This Anti-Feature is applied to apps that track you and/or report your activity to somewhere, either without your permission or by default (i.e. **you’d have to actively seek out an option to disable it**).
> Examples of where it would not be applied - any of the above, if the functionality is opt-in (i.e. you are asked before it happens) and disabled by default. Enabling it should then also require informed consent, i.e. requiring a privacy policy similar to GDPR, and avoid collecting personal data (PII) as far as possible.

https://f-droid.org/en/docs/Anti-Features/#Tracking

As I said in the OP, currently they are OPT-OUT, right?

**Comment by bassosimone** [comment]
@licaon-kter I understand that the HTTPS connections happen before tapping on "Let's Go", correct?

**Comment by lorenzoPrimi** [comment]
They are OPT-IN, they are off by default and enabled only in the moment you press "Let's go"
We don't track or report anything until there is your permission to do so, the only think that happens at app opening is init the SDKs with no permission to report anything.

**Comment by licaon-kter** [comment]
I don't know, or really care, think of it this way: I _(some anon from the 'net but with past interactions with F-Droid merged apps and checking them)_ DID NOT UNDERSTOOD the implications of your onboarding.

**Comment by licaon-kter** [comment]
> `To improve OONI Probe we also collect app usage metrics and crash reports. Learn more on some site`

Ok, no indication here that I CAN DISABLE this.

> `Change defaults`

Of what? The app? I don't know if I want to... I just barely installed it, dunno how it works or if I need to.

This whole page looks more like a "[Dark pattern](https://en.wikipedia.org/wiki/Dark_pattern)"

> They are OPT-IN, they are off by default and enabled only in the moment you press "Let's go"

Defaults means I DO NOT CHANGE ANYTHING, your button changes something hidden from FALSE to TRUE? O RLY?

> We don't track or report anything until there is your permission to do so

You did not ask for permission to begin with.

**Comment by bassosimone** [comment]
> [...] DID NOT UNDERSTOOD the implications of your onboarding.

If the onboarding is not clear enough with respect its implications, then this is a problem that we should address.

Thank you so much @licaon-kter for opening this issue and explaining this problem to us!

We will discuss this internally and come up with a proposed solution to improve clarity and make sure we don't open HTTPS connections until we have permission.

**Comment by lorenzoPrimi** [comment]
I can understand they seems messy, maybe the way we communicate is not clear enough. 
Let me explain better this and you can tell us how we can improve the communication.

The defaults settings on the app as you can see from the file are no crash and not analytics. This is due to the GDPR regulation.
Then we ask the user to accept are the settings you see in that onboarding screen: so yes submit measurement, submit crash, submit analytics. If the users tap's let's go the last two will be changed in true.
If the users taps change settings these will stay false.

I understand it can be confusing, especially because "Let's go" doesn't mean much, as @bassosimone said we will check how this can be improved, feel free to leave your feedbacks

**Comment by licaon-kter** [comment]
Ping me when you have some changes.

**Comment by hellais** [comment]
@licaon-kter thanks again for reporting this problem!

In order to address this we have taken the following steps:

1. Fixed on all platforms the issue of registering with the crash reporting SDKs prior to the informed consent procedure
2. Disabled app usage metrics entirely across all platforms
3. Disabled crash reporting on all f-droid builds

We are going to ship these changes this week inside of OONI Probe 2.10.1.

In future releases we also plan to improve our informed consent procedure to make it more clear that the user is opting into crash reporting with the last step of the onboarding, see: https://github.com/ooni/probe/issues/1398.

Do let us know if you have any questions and we hope this helps in addressing the important concerns that you raised.

Have a great day!

**Comment by hellais** [comment]
@licaon-kter I see that OONI Probe Mobile 2.10.1 has been uploaded to the F-Droid repository. As mentioned this version of OONI Probe no longer includes any sort of crash reporting or app usage metric functionality.

Would it be possible to remove the tracking anti-feature from the F-Droid repository listing?

Thanks!

**Comment by licaon-kter** [comment]
Yes, just a question of time

**Comment by hellais** [comment]
Thanks a bunch!

**Comment by licaon-kter** [comment]
Fyi https://gitlab.com/fdroid/fdroiddata/-/merge_requests/9203

**Comment by bassosimone** [comment]
🙏 !
